compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt

```diff
@@ -0,0 +1,144 @@
+import androidx.compose.material3.adaptive.layout.PaneAdaptedValue
+import androidx.compose.material3.adaptive.layout.SupportingPaneScaffold
+import androidx.compose.material3.adaptive.layout.SupportingPaneScaffoldRole
+import androidx.compose.material3.adaptive.layout.ThreePaneScaffoldRole
+import androidx.compose.material3.adaptive.layout.ThreePaneScaffoldScope
+import androidx.compose.material3.adaptive.navigation.rememberSupportingPaneScaffoldNavigator
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+
+@Composable
+fun SampleSupportingPaneScaffoldParts() {
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_nav_and_back]
+    val navigator = rememberSupportingPaneScaffoldNavigator()
+
+    BackHandler(navigator.canNavigateBack()) {
+        navigator.navigateBack()
+    }
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_nav_and_back]
+
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_params]
+    SupportingPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        mainPane = { /*...*/ },
+        supportingPane = { /*...*/ },
+    )
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_params]
+}
+
+@Composable
+fun SampleSupportingPaneScaffoldFull() {
+    // [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_full]
+    val navigator = rememberSupportingPaneScaffoldNavigator()
+
+    BackHandler(navigator.canNavigateBack()) {
+        navigator.navigateBack()
+    }
+
+    SupportingPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        mainPane = {
+            AnimatedPane(modifier = Modifier.safeContentPadding()) {
+                // Main pane content
+                if (navigator.scaffoldValue.secondary == PaneAdaptedValue.Hidden) {
+                    Button(
+                        modifier = Modifier.wrapContentSize(),
+                        onClick = {
+                            navigator.navigateTo(SupportingPaneScaffoldRole.Supporting)
+                        }
+                    ) {
+                        Text("Show supporting pane")
+                    }
+                } else {
+                    Text("Supporting pane is shown")
+                }
+            }
+        },
+        supportingPane = {
+            AnimatedPane(modifier = Modifier.safeContentPadding()) {
+                // Supporting pane content
+                Text("Supporting pane")
+            }
+        },
+    )
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_full]
+}
+
+// [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_extracted_panes]
+@Composable
+fun ThreePaneScaffoldScope.MainPane(
+    shouldShowSupportingPaneButton: Boolean,
+    onNavigateToSupportingPane: () -> Unit,
```

Comment:
Should this have a `modifier: Modifier = Modifier` for completeness?

Reply:
Good call, yeah let me do that to reinforce best practices